Source/WebKit/NetworkProcess/storage/QuotaManager.cpp

```diff
@@ -44,6 +44,14 @@ QuotaManager::QuotaManager(uint64_t quota, GetUsageFunction&& getUsageFunction,
     ASSERT(m_quota);
 }
 
+uint64_t QuotaManager::usage()
+{
+    if (!m_usage)
+        m_usage = m_getUsageFunction();
```

Comment:
I don't quite yet why it is OK to cache m_usage? Where do we invalidate m_usage if usage changes?

Reply:
We will update m_usage during quota check (when usage is bigger than quota).
But it seems we relied on m_usage is null to decide if we are doing the first quota check and update initial quota based on that, so I guess we'd better not cache it here. Will change.